src/tasks/MotionForceTask.cpp

```diff
@@ -164,7 +165,12 @@ void MotionForceTask::initialSetup() {
 		_current_position, _current_orientation, getLoopTimestep());
 	enableInternalOtgAccelerationLimited(0.3, 1.0, M_PI / 3, M_PI);
 
-	reInitializeTask();
+	// singularity handling
+	MatrixXd J_posture = getConstRobotModel()->linkDependency(_link_name);
```

Comment:
I still don't get why we need this jacobian and why we cannot just take the "normal" jacobian